packages/core/contracts/zodiac/OptimisticGovernor.sol

```diff
@@ -212,59 +205,53 @@ contract OptimisticGovernor is Module, Lockable {
+        optimisticOracle.setCustomLiveness(identifier, time, "", liveness);
 
         // Get the bond from the proposer and approve the bond and final fee to be used by the oracle.
         // This will fail if the proposer has not granted the OptimisticGovernor contract an allowance
         // of the collateral token equal to or greater than the totalBond.
         collateral.safeTransferFrom(msg.sender, address(this), totalBond);
         collateral.safeIncreaseAllowance(address(optimisticOracle), totalBond);
 
-        optimisticOracle.proposePriceFor(
-            msg.sender,
-            address(this),
-            identifier,
-            time,
-            ancillaryData,
-            PROPOSAL_VALID_RESPONSE
-        );
+        optimisticOracle.proposePriceFor(msg.sender, address(this), identifier, time, "", PROPOSAL_VALID_RESPONSE);
 
         uint256 challengeWindowEnds = time + liveness;
 
-        emit TransactionsProposed(id, proposer, time, proposal, _explanation, challengeWindowEnds);
+        emit TransactionsProposed(proposer, time, proposal, _explanation, challengeWindowEnds);
     }
 
     /**
      * @notice Executes an approved proposal.
-     * @param _proposalId the id of the proposal being executed.
      * @param _transactions the transactions being executed. These must exactly match those that were proposed.
-     * @param _originalTime the timestamp of the original proposal.
      */
-    function executeProposal(
-        uint256 _proposalId,
-        Transaction[] memory _transactions,
-        uint256 _originalTime
-    ) external payable nonReentrant {
+    function executeProposal(Transaction[] memory _transactions) external payable nonReentrant {
         // Recreate the proposal hash from the inputs and check that it matches the stored proposal hash.
-        uint256 id = _proposalId;
-
-        // Construct the ancillary data.
-        bytes memory ancillaryData = AncillaryData.appendKeyValueUint("", "id", id);
+        bytes32 _proposalHash = keccak256(abi.encode(_transactions));
 
         // This will reject the transaction if the proposal hash generated from the inputs does not match the stored proposal hash.
-        require(proposalHashes[id] == keccak256(abi.encode(_transactions)), "proposal hash does not match");
+        require(proposalHashes[_proposalHash] != 0, "proposal hash does not exist");
+
+        // Get the original proposal time.
+        uint256 _originalTime = proposalHashes[_proposalHash];
 
         // Remove proposal hash so transactions can not be executed again.
-        delete proposalHashes[id];
+        delete proposalHashes[_proposalHash];
 
         // This will revert if the price has not been settled and can not currently be settled.
-        int256 price = optimisticOracle.settleAndGetPrice(identifier, _originalTime, ancillaryData);
+        int256 price = optimisticOracle.settleAndGetPrice(identifier, _originalTime, "");
```

Comment:
If we want to ensure that this contract never allows disputed proposals to be used for settlement, we should call check that getRequest shows no disputer.
```
require(optimisticOracle.getRequest(address(this), identifier, _originalTime, ancillaryData).disputer == address(0), "Must call deleteRejectedProposal instead");
```